dace/sdfg/validation.py

```diff
@@ -191,6 +191,22 @@ def validate_state(state: 'dace.sdfg.SDFGState',
         except Exception as ex:
             raise InvalidSDFGNodeError("Node validation failed: " + str(ex), sdfg, state_id, nid) from ex
 
+        if Config.get_bool('check_race_conditions'):
+            if isinstance(node, nd.AccessNode):
+                incoming_edge_memlet_ranges = []
+                for e in state.in_edges(node):
+                    memlet_range = str(e.data.subset).split(':')
```

Comment:
String based analysis can be a bit unsafe, especially when all the information can already be retrieved from the `subsets.Range` class (which is the type of `e.data.subset`)